source/cdk/lib/msam-browser-app.ts

```diff
@@ -0,0 +1,357 @@
+            }
+        );
+
+        // Browser App S3 Bucket
+        const s3BucketBrowserApp = new s3.Bucket(this, 'MSAMBrowserAppBucket', {
+            cors: [
+                {
+                    allowedHeaders: ['*'],
+                    allowedMethods: [s3.HttpMethods.GET],
+                    allowedOrigins: ['*'],
+                }
+            ],
+            encryption: s3.BucketEncryption.S3_MANAGED,
+        });
+
+        this.MSAMBrowserAppBucket = Fn.join('', [
+            'https://s3.console.aws.amazon.com/s3/buckets/',
+            s3BucketBrowserApp.bucketName,
+            '/?region=',
+            Aws.REGION,
+        ]);
+
+        utils.setNagSuppressRules(s3BucketBrowserApp,
+            {
+                id: 'AwsSolutions-S2',
+                reason: 'Public access blocked by default'
+            },
+            {
+                id: 'W51',
+                reason: 'This is the logging bucket. No bucket policy needed.'
+            },
+            {
+                id: 'AwsSolutions-S10',
+                reason: 'HTTPS requirement not supported at this time.',
+            }
+        );
+
+        (s3BucketBrowserApp.node.defaultChild as s3.CfnBucket).loggingConfiguration = {
+            destinationBucketName: s3BucketbrowserAppLogs.bucketName,
+            logFilePrefix: 'app-bucket/',
+        };
+
+        // CloudFront OriginAccessIdentity
+        const cloudfrontOriginAccessIdentity = new cloudfront.OriginAccessIdentity(this, 'MSAMAppBucketOriginAccessIdentity', {
+            comment: Fn.join('', [
+                'Origin Access Identity for ',
+                s3BucketBrowserApp.bucketName,
+            ])
+        });
+
+        // Browser App S3 Bucket Policy
+        const s3BucketAppPolicy = this.createS3BucketPolicy('MSAMBrowserAppBucketPolicy', {
+            bucket: s3BucketBrowserApp.bucketName,
+            policyDocument: new iam.PolicyDocument({
+                statements: [
+                    new iam.PolicyStatement({
+                        effect: iam.Effect.ALLOW,
+                        principals: [new iam.CanonicalUserPrincipal(cloudfrontOriginAccessIdentity.cloudFrontOriginAccessIdentityS3CanonicalUserId)],
+                        actions: ['s3:GetObject'],
+                        resources: [
+                            Fn.join('', [
+                                s3BucketBrowserApp.bucketArn,
+                                '/*',
+                            ]),
+                        ],
+                    }),
+                ],
+            }),
+        });
+
+        utils.setNagSuppressRules(s3BucketAppPolicy,
+            {
+                id: 'AwsSolutions-S10',
+                reason: 'HTTPS requirement not supported at this time.',
+            }
+        )
+
+        // CloudFront Distribution
+        const cloudfrontDistribution = new cloudfront.CfnDistribution(this, 'MSAMAppBucketCloudFrontDistribution', {
```

Comment:
Yes, there were many required properties that are not supported in the L2 construct that it would make the code much cleaner to simply use the L1 construct instead of modifying the underlying resource object

Reply:
Okay, that makes sense.

One suggestion: It might be helpful to include a code comment explaining that just so it's clear that the L1 construct was intentionally chosen for a reason.